packages/core/src/oidc/init.ts

```diff
@@ -198,6 +209,63 @@ export default function initOidc(envSet: EnvSet, queries: Queries, libraries: Li
       },
     },
     extraParams: [OIDCExtraParametersKey.InteractionMode],
+    extraTokenClaims: async (ctx, token) => {
+      try {
+        const { isDevFeaturesEnabled, isCloud } = EnvSet.values;
+        // No cloud connection for OSS version, skip.
+        if (!isDevFeaturesEnabled || !isCloud) {
+          return;
+        }
+
+        const isTokenClientCredentials = token instanceof ctx.oidc.provider.ClientCredentials;
+
+        const { script, envVars } =
+          (await trySafe(
+            logtoConfigs.getJwtCustomizer(
+              isTokenClientCredentials
+                ? LogtoJwtTokenKey.ClientCredentials
+                : LogtoJwtTokenKey.AccessToken
+            )
+          )) ?? {};
+
+        if (!script) {
+          return;
+        }
+
+        // Wait for cloud API to be ready and we can use cloud connection client to request the API.
+        const client = await cloudConnection.getClient();
+
+        // We pass context to the cloud API only when it is a user's access token.
+        const logtoUserInfo = conditional(
+          !isTokenClientCredentials &&
+            token.accountId &&
+            (await libraries.jwtCustomizers.getUserContext(token.accountId))
+        );
+        /**
+         * `token` and `context` can not be assigned to Record<string, Json> according to the type inference,
+         * use request body guard to ensure the type.
+         *
+         * Use direct type casting to avoid the type inference issue since if the type is not correct the client
+         * will throw an Zod type error, there is no need to implement the zod guard and error handling here.
+         */
+        // eslint-disable-next-line no-restricted-syntax
+        const payload = {
+          script,
+          envVars,
+          token,
+          context: conditional(logtoUserInfo && { user: logtoUserInfo }),
+        } as unknown as CustomJwtFetcher;
```

Comment:
avoid using as unknown as CustomJwtFetcher?

Reply:
Linter hint shows `AccessToken|ClientCredentials` can not assigned to `Json`, while we use `Json` instead of `unknown` in `@withtyped`. It is OK to do this since the Cloud service can handle this.
<img width="970" alt="image" src="https://github.com/logto-io/logto/assets/15182327/0d5780f8-e52a-4999-a9f8-ec3bbf0ebf31">